kola/tests/crio/crio.go

```diff
@@ -416,3 +417,28 @@ func testCrioInfo(c cluster.TestCluster) {
 	}
 
 }
+
+// crioPodContinuesDuringServiceRestart verifies that a crio pod does not
+// stop when the service is restarted
+func crioPodContinuesDuringServiceRestart(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	crioConfigPod, crioConfigContainer, err := genContainer(
+		c, m, "restart-test", "sleep", []string{"sleep"}, []string{"sleep"})
+	if err != nil {
+		c.Fatal(err)
+	}
+	cmdCreatePod := fmt.Sprintf("sudo crictl runp %s", crioConfigPod)
+	podID := c.MustSSH(m, cmdCreatePod)
+	containerID := c.MustSSH(m, fmt.Sprintf("sudo crictl create %s %s %s",
+		podID, crioConfigContainer, crioConfigPod))
+	// Set the restart to occur in 15 seconds
+	c.MustSSH(m, "sleep 15 && sudo systemctl restart crio &")
```

Comment:
I don't think this is working properly, I added some code to test it like so:

```golang
before := time.Now()
c.MustSSH(m, "sleep 15 && sudo systemctl restart crio &")
fmt.Println(time.Since(before))
```

which is printing that it waited for at least 15s before exiting `c.MustSSH`.

One alternative idea would be to build both `sleep` & `echo` into the container and have the container mount in a folder and write to it after the wait. Then the code inside of the test could be something like:

```golang
c.MustSSH(m, "sudo crictl exec %s <sleep & write to file>")
time.Sleep(15 * time.Second())
c.MustSSH(m, "sudo systemctl restart crio")
time.Sleep(45 * time.Second())
output := c.MustSSH(m, "cat <file>")
```

Reply:
Thanks @arithx, I'll give it a go!

Reply:
Yeah, that suggestion does sound simpler.

Another approach would be running `MustSSH` in a goroutine.  Yet another would be to use `systemd-run` to create a background task on the target host.